.travis.yml

```diff
@@ -1,37 +1,38 @@
 language: scala
 
+# These set $TRAVIS_JDK_VERSION which is used, in turn, to get a
+# version of AdoptOpenJDK with the install-jdk.sh command.
+jdk:
+  - openjdk8
+  - openjdk11
+
 scala:
   - 0.18.1-RC1
   - 2.12.10
   - 2.13.1
 
 env:
-  - SCALAJS_VERSION=         ADOPTOPENJDK=8
-  - SCALAJS_VERSION=0.6.31   ADOPTOPENJDK=8
-  - SCALAJS_VERSION=1.0.0-M8 ADOPTOPENJDK=8
-  - SCALAJS_VERSION=         ADOPTOPENJDK=11
+  # The empty SCALAJS_VERSION will only compile for the JVM
+  - SCALAJS_VERSION=
+  - SCALAJS_VERSION=0.6.31
+  - SCALAJS_VERSION=1.0.0-M8
 
 matrix:
   exclude:
+    - jdk: openjdk11
+      env: SCALAJS_VERSION=0.6.29
+    - jdk: openjdk11
+      env: SCALAJS_VERSION=1.0.0-M8
     - scala: 0.18.1-RC1
-      env: SCALAJS_VERSION=0.6.31   ADOPTOPENJDK=8
+      env: SCALAJS_VERSION=0.6.31
     - scala: 0.18.1-RC1
-      env: SCALAJS_VERSION=1.0.0-M8 ADOPTOPENJDK=8
-
-before_install:
-  # adding $HOME/.sdkman to cache would create an empty directory, which interferes with the initial installation
-  - "[[ -d $HOME/.sdkman/bin ]] || rm -rf $HOME/.sdkman/"
-  - curl -sL https://get.sdkman.io | bash
-  - echo sdkman_auto_answer=true > $HOME/.sdkman/etc/config
-  - source "$HOME/.sdkman/bin/sdkman-init.sh"
-
-install:
-  - sdk install java $(sdk list java | grep -o "$ADOPTOPENJDK\.[0-9\.]*hs-adpt" | head -1)
-  - unset JAVA_HOME
-  - java -Xmx32m -version
-  - git fetch --tags # get all tags for sbt-dynver
+      env: SCALAJS_VERSION=1.0.0-M8
 
-script: ./build.sh
+script:
+  - export JAVA_HOME=$HOME/$TRAVIS_JDK_VERSION
```

Comment:
Is `JAVA_HOME` necessary here? I avoid it because having `JAVA_HOME` causes subtle difference in the way sbt behaves with Java code.

Reply:
To be honest, I'm just following directions:

https://docs.travis-ci.com/user/languages/java/#switching-jdks-to-java-10-and-up-within-one-job

I presume these incantations are needed because how install-jdk.sh works, but I don't know.